pkg/resourcegroup/controllers/typeresolver/typeresolver_test.go

```diff
@@ -51,3 +57,211 @@ func TestResolve(t *testing.T) {
 		})
 	}
 }
+
+func TestRefresh(t *testing.T) {
```

Comment:
Mostly lgtm, nit missing a case for non-errGroupDiscoveryFailed error?

Reply:
There's a test case in there for "other errors"

Reply:
oh, i see what you mean. I'll add another

Reply:
added.